### PR TITLE
Adds LectureListServlet

### PR DESCRIPTION
**Pull request by jaysaleh**
Moves getting the lecture list to a different servlet.

### CODE REVIEW

File: src/main/java/com/googleinterns/zoomtube/servlets/LectureListServlet.java [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+import java.util.ArrayList;
+import java.util.List;
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+/** Returns list of Lectures. */
+public class LectureListServlet extends HttpServlet {
+  private static DatastoreService datastore;
```
Comment (n-wach) [inline src/main/java/com/googleinterns/zoomtube/servlets/LectureListServlet.java:35]: I think this shouldn't be static

File: src/main/java/com/googleinterns/zoomtube/servlets/LectureListServlet.java [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+    response.setContentType("application/json");
+    response.getWriter().println(gson.toJson(lectures));
+  }
+
+  /** Returns lectures stored in the database. */
+  private List<Lecture> getLectures() {
+    Query query = new Query(LectureUtil.KIND);
+    PreparedQuery results = datastore.prepare(query);
+    List<Lecture> lectures = new ArrayList<>();
+    for (Entity lecture : results.asIterable()) {
```
Comment (n-wach) [inline src/main/java/com/googleinterns/zoomtube/servlets/LectureListServlet.java:55]: call this `lectureEntity` and the new one below `lecture`?

File: src/main/java/com/googleinterns/zoomtube/servlets/LectureListServlet.java [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    List<Lecture> lectures = getLectures();
+    Gson gson = new Gson();
+    response.setContentType("application/json");
+    response.getWriter().println(gson.toJson(lectures));
+  }
+
+  /** Returns lectures stored in the database. */
+  private List<Lecture> getLectures() {
```
Comment (amuamushu) [inline src/main/java/com/googleinterns/zoomtube/servlets/LectureListServlet.java:51]: nit: you can just call this `lectures()`. I think it's clear that you're getting the lectures based on the return type.

File: src/main/webapp/lectures/lectures.js [hunk 4]
```diff
@@ -22,7 +22,7 @@ loadLectureList();
 
 /** Fetches data from servlet and sets it in the lecture selection page. */
```
Comment (amuamushu) [inline src/main/webapp/lectures/lectures.js:23]: nit: can you indicate the name of the servlet in the method header.

File: src/test/java/com/googleinterns/zoomtube/servlets/LectureListServletTest.java [hunk 5]
```diff
@@ -0,0 +1,110 @@
...
+
+    String json = content.toString();
+    assertThat(json).startsWith("[]");
+  }
+
+  @Test
+  public void doGet_oneLectureInDatabase_shouldReturnOneLecture() throws IOException {
+    when(request.getParameter(LINK_INPUT)).thenReturn(TEST_LINK);
+    datastoreService.put(LectureUtil.createEntity(/* lectureName= */ "", TEST_LINK, TEST_ID));
+    StringWriter content = new StringWriter();
```
Comment (billie-cat) [inline src/test/java/com/googleinterns/zoomtube/servlets/LectureListServletTest.java:97]: These could be declared at the class level, given they are used in multiple tests.

File: src/main/webapp/lectures/lectures.js [hunk 6]
```diff
@@ -20,9 +20,12 @@ const REDIRECT_PAGE = '/view/';
 
 loadLectureList();
 
-/** Fetches data from servlet and sets it in the lecture selection page. */
+/**
+ * Fetches data from LectureListServlet and sets it in the lecture selection
```
[the remark this answers is not in the document]
Comment (jaysaleh) [reply src/main/webapp/lectures/lectures.js:24]: It is fetching it from LectureListServlet which gets them from Datastore, should I replace LectureListServlet with Datastore?

File: src/test/java/com/googleinterns/zoomtube/servlets/LectureListServletTest.java (hunk 5 above)
Comment (jaysaleh) [reply src/test/java/com/googleinterns/zoomtube/servlets/LectureListServletTest.java]: fixed

File: src/test/java/com/googleinterns/zoomtube/servlets/LectureListServletTest.java [hunk 7]
```diff
@@ -0,0 +1,111 @@
...
+  @Mock private HttpServletResponse response;
+
+  private final LocalServiceTestHelper testServices =
+      new LocalServiceTestHelper(new LocalDatastoreServiceTestConfig());
+  private DatastoreService datastoreService;
+  private LectureListServlet servlet;
+
+  /* Writer where response is written. */
+  StringWriter content;
+  PrintWriter writer;
```
Comment (f-camp) [inline src/test/java/com/googleinterns/zoomtube/servlets/LectureListServletTest.java:66]: Can these be private?

Comment (f-camp) [inline src/test/java/com/googleinterns/zoomtube/servlets/LectureListServletTest.java:66]: Can these be private?

File: src/main/java/com/googleinterns/zoomtube/servlets/LectureServlet.java [hunk 8]
```diff
@@ -80,10 +80,7 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
 
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    List<Lecture> lectures = getLectures();
-    Gson gson = new Gson();
-    response.setContentType("application/json");
-    response.getWriter().println(gson.toJson(lectures));
+    // TODO: Implement getting one lecture.
```
Comment (n-wach) [inline src/main/java/com/googleinterns/zoomtube/servlets/LectureServlet.java:83]: even tho this is missing, we're OK to merge b/c nothing uses this... correct?

File: src/main/webapp/lectures/lectures.js [hunk 9]
```diff
@@ -20,9 +20,12 @@ const REDIRECT_PAGE = '/view/';
...
 loadLectureList();
 
-/** Fetches data from servlet and sets it in the lecture selection page. */
+/**
+ * Fetches avaiable Lectures from LectureListServlet and sets them in the
+ * lecture selection page.
+ */
 async function loadLectureList() {
-  const response = await fetch('/lecture');
+  const response = await fetch('/lecture-list');
```
Comment (n-wach) [inline src/main/webapp/lectures/lectures.js:28]: can we turn this into `ENDPOINT_LECTURE_LIST`, and include the constant in the function comment?

File: src/main/java/com/googleinterns/zoomtube/servlets/LectureServlet.java (hunk 8 above)
Comment (jaysaleh) [reply src/main/java/com/googleinterns/zoomtube/servlets/LectureServlet.java:83]: Yes

**Pull request merged**